upgrader.go

```diff
@@ -460,18 +465,24 @@ func (u *Upgrader) getLaunchTemplateForASG(asgName string) (*ec2types.LaunchTemp
 	return lt, ltv.LaunchTemplateVersions[0].LaunchTemplateData, nil
 }
 
-func (u *Upgrader) getImageByID(imageID string) (ec2types.Image, error) {
+func (u *Upgrader) getImageByID(imageID string, filters ...string) (ec2types.Image, error) {
 	imgInput := &ec2.DescribeImagesInput{
 		ImageIds: []string{
 			imageID,
 		},
 	}
+	if len(filters) > 0 {
+		imgInput.Filters = []ec2types.Filter{{
+			Name:   aws.String("name"),
```

Comment:
Name seems confusing here, but that's Amazon's fault.

Reply:
yep, leave it to Amazon to make things complicated